data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraphData.py

```diff
@@ -0,0 +1,174 @@
+    parser._action_groups.pop()
+    required = parser.add_argument_group('Required arguments')
+    optional = parser.add_argument_group('Optional arguments')
+
+    required.add_argument('-d','--data', type=str, metavar='', required=True, help="Path to input-data used by percolator.")
+    optional.add_argument('-py','--python_version', type=str, metavar='', default="python3" , required=False, help="Python version to call other python modules from.")
+    optional.add_argument('-r','--runs', type=int, metavar='', default=3, required=False, help="Number of tests to evaluate.")
+    optional.add_argument('-f','--flags', type=str, default = "" , metavar='', required=False, help="Flags to use with the executable.")
+    optional.add_argument('-a','--await_tests', default=False, action="store_true", required=False, help="Wait for others processes running this script to finish before starting.")
+    optional.add_argument('-c','--comments', type=str, default="", metavar='', required=False, help="Comments regarding test details.")
+    optional.add_argument('-fname','--filename', type=str, default="barFigure", metavar='', required=False, help="Name of produced charts.")
+    optional.add_argument('-xstep','--x_step_size', type=int, default=50000, metavar='', required=False, help="Amount of input lines that also determine the step size between tests.")
+    return parser.parse_args()
+
+def makeReducedFile(fileNameOriginal, fileNameReduced, numLines):
+    if Path(fileNameReduced).is_file():
+        return
+    fileReduced = open(fileNameReduced, "w")
+    counter = 0
+    with open(fileNameOriginal) as f:
+        for line in f:
+            counter += 1
+            if counter > numLines:
+                break
+            fileReduced.write(line)
+    fileReduced.close()
+
+def countLinesInFile(filename): ####https://stackoverflow.com/a/27518377/7202012
+    f = open(filename, 'rb')
+    bufgen = takewhile(lambda x: x, (f.raw.read(1024*1024) for _ in repeat(None)))
+    return sum( buf.count(b'\n') for buf in bufgen )
+
+def getTemporaryInDataFileName(number, dataFilename):
+    return dataFilename + str(number) + ".tab"
+
+def getTerminalCommand(args, data):
+    result = args.python_version
+    result += " " + testScriptPath
+    result += " -d " + data
+    if len(args.flags) > 0:
+        result += " -f \"" + args.flags + "\""
+    result += " -r " + str(args.runs)
+    if args.await_tests is True:
+        result += " -a "
+    print("RES: " + result)
+    return result
+
+def generateInputFiles(args):
+    maxNumLines = countLinesInFile(args.data)
+    dataFilename = Path(args.data).stem
+    numDataFiles = 0
+    for i in range(0, int(maxNumLines/minNumLines)):
+        numDataFiles += 1
+        fileNumber = minNumLines*numDataFiles
+        makeReducedFile(args.data, pathToOutputData + "/" + getTemporaryInDataFileName(fileNumber, dataFilename), fileNumber)
```

Comment:
The code is rather inconsistent with how it creates output strings, sometimes using str.format() and sometimes building them up manually with str() and concatenation. (Compare this line with line 100 for example -- they're doing the same thing but in two different ways.) Consistency makes the code easier to read, so it would be nice if the file would pick one of the two methods and stuck with it.